int32 index
string value
---
int32 res